transforms/universal/fdedup/python/src/get_duplicate_list_transform.py

```diff
@@ -0,0 +1,168 @@
+
+# command line arguments
+subfolder_cli_param = f"{cli_prefix}{subfolder_key}"
+""" The name of the subfolder with the duplicate records"""
+consolidated_filename_cli_param = f"{cli_prefix}{consolidated_filename_key}"
+""" The name of the file with the consolidated list of duplicates"""
+
+captured_arg_keys = [
+    subfolder_key,
+    consolidated_filename_key,
+]
+
+# defaults
+subfolder_default = "docs_to_remove"
+""" Default name of the subfolder with the duplicate records"""
+consolidated_filename_default = os.path.join("docs_to_remove_consolidated", "docs_to_remove_consolidated.parquet")
+""" Default name of the file with the consolidated list of duplicates"""
+
+
+class GetDuplicateListTransform(AbstractFolderTransform):
+    """
+    This is an intermediate step of the fuzzy dedup pipeline. It runs in a single
+    location and consolidates in a single file all the duplicates found for each
+    band segment.
+    Args:
+        subfolder: name of the subfolder with the duplicate records
+        consolidated_filename: name of the file with the consolidated list of duplicates
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments
+        defined by the companion runtime, ClusterAnalysisTransformRuntime.
+        """
+        super().__init__(config)
+        self.subfolder = config.get(subfolder_key, subfolder_default)
+        self.consolidated_filename = config.get(consolidated_filename_key, consolidated_filename_default)
+        self.data_access = config.get("data_access")
+        self.logger = get_logger(__name__)
+
+    def transform(self, folder_name: str) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        self.logger.info(f"Get Duplicate List for folder {folder_name}")
+        metadata = {}
+        input_folder = self.sanitize_folder_name(os.path.join(self.data_access.input_folder, folder_name))
+        files, retries = self.data_access.get_folder_files(
+            path=input_folder,
+            extensions=[".parquet"],
+            return_data=True,
+        )
+        if retries > 0:
+            metadata |= {"data_access_retries": retries}
+        output_folder = self.sanitize_folder_name(self.data_access.output_folder)
+        output_path = os.path.join(output_folder, self.consolidated_filename)
+
+        # consolidate into a single data frame band hashes computed by workers
+        consolidated_dataframe, consolidation_stats = self.consolidate_docs_to_remove_files(files)
+        self.logger.info(f"{len(consolidated_dataframe)} documents marked as duplicates")
+        metadata |= consolidation_stats
+        output_data = TransformUtils.convert_arrow_to_binary(consolidated_dataframe.to_arrow())
+        return [(output_data, output_path)], metadata
+
+    def sanitize_folder_name(self, folder_name: str) -> str:
```

Comment:
I would recommend making all methods other than transform() and flush() as protected or private with _ or __ prefixes.